examples/mnist_00.py

```diff
@@ -0,0 +1,168 @@
+                    epoch,
+                    batch_idx * len(data),
+                    len(train_loader.dataset),
+                    100.0 * batch_idx / len(train_loader),
+                    loss.item(),
+                )
+            )
+            if args.dry_run:
+                break
+
+
+def test(model, device, test_loader):
+    model.eval()
+    test_loss = 0
+    correct = 0
+    with torch.no_grad():
+        for data, target in test_loader:
+            data, target = data.to(device), target.to(device)
+            output = model(data)
+            test_loss += F.nll_loss(
+                output, target, reduction="sum"
+            ).item()  # sum up batch loss
+            pred = output.argmax(
+                dim=1, keepdim=True
+            )  # get the index of the max log-probability
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= len(test_loader.dataset)
+
+    print(
+        "\nTest set: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n".format(
+            test_loss,
+            correct,
+            len(test_loader.dataset),
+            100.0 * correct / len(test_loader.dataset),
+        )
+    )
+
+
+@hydra.main(config_name="mnistconf")
+def main(cfg):  # DIFF
+    print(cfg.pretty())
+    use_cuda = not cfg.no_cuda and torch.cuda.is_available()  # DIFF
+    torch.manual_seed(cfg.seed)  # DIFF
+    device = torch.device("cuda" if use_cuda else "cpu")
+
+    train_kwargs = {"batch_size": cfg.batch_size}  # DIFF
+    test_kwargs = {"batch_size": cfg.test_batch_size}  # DIFF
+    if use_cuda:
+        cuda_kwargs = {"num_workers": 1, "pin_memory": True, "shuffle": True}
+        train_kwargs.update(cuda_kwargs)
+        test_kwargs.update(cuda_kwargs)
+
+    transform = transforms.Compose(
+        [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]
+    )
+    dataset1 = datasets.MNIST("../data", train=True, download=True, transform=transform)
+    dataset2 = datasets.MNIST("../data", train=False, transform=transform)
+    train_loader = torch.utils.data.DataLoader(dataset1, **train_kwargs)
+    test_loader = torch.utils.data.DataLoader(dataset2, **test_kwargs)
+
+    model = Net().to(device)
+
+    optimizer = Adadelta(
+        lr=cfg.adadelta.lr,
+        rho=cfg.adadelta.rho,
+        eps=cfg.adadelta.eps,
+        weight_decay=cfg.adadelta.weight_decay,
+        params=model.parameters(),
+    )  # DIFF
+    scheduler = StepLR(
+        step_size=cfg.steplr.step_size,
+        gamma=cfg.steplr.gamma,
+        last_epoch=cfg.steplr.last_epoch,
+        optimizer=optimizer,
+    )  # DIFF
+
+    for epoch in range(1, cfg.epochs + 1):  # DIFF
+        train(cfg, model, device, train_loader, optimizer, epoch)  # DIFF
+        test(model, device, test_loader)
+        scheduler.step()
+
+    if cfg.save_model:  # DIFF
+        torch.save(model.state_dict(), cfg.checkpoint_name)  # DIFF
```

Comment:
worth commenting that the checkpoint is saved to an automatically generated working directory here.